scripts/generate-app-config/generateContentConfig.js

```diff
@@ -0,0 +1,225 @@
+  const menuItems = acc.menuItems || [];
+  const hasSubMenu = page.pages && page.pages.length > 0;
+  const firstItemsHasSubMenu = hasSubMenu && page.pages[0].pages;
+  const subMenuHasMoreThanOneItem = hasSubMenu && page.pages.length > 1;
+  const isRootItem = !rootPath;
+
+  const routePath = `${rootPath}${page.path}`;
+  let descendantMenuItems;
+
+  // If the given page, has sub menu items, add them to the overall route object.
+  if (hasSubMenu) {
+    // Recursively call to get child content, and menu items
+    const { content: childContent, menuItems: childMenuItems } = getPageContentConfig(page.pages, routePath);
+
+    content = Object.assign(content, childContent);
+    descendantMenuItems = childMenuItems;
+  }
+
+  if (!isRootItem || firstItemsHasSubMenu || subMenuHasMoreThanOneItem) {
+    // Provide the menu item for this content page.
+    menuItems.push(menuItem(page.name, routePath, hasSubMenu, descendantMenuItems));
+  }
+
+  if (page.content) {
+    content[routePath] = contentRouteItem(page.name, routePath, { contentPath: page.content }, page.props, page.type);
+  } else if (isRootItem && !firstItemsHasSubMenu) {
+    content[routePath] = redirectRouteItem(page.name, routePath, descendantMenuItems[0].path);
+  }
+
+  return { content, menuItems };
+}, { content: {}, menuItems: [] });
+
+/**
+* Create a page config object for the top level page.
+*/
+const getPageConfig = (name, pagePath, pages, type, siteConfig) => {
+  const { readMeContent } = siteConfig;
+  const config = {
+    name: startCase(name),
+    path: pagePath,
+    pages,
+  };
+
+  // Special logic to add a home component with a readme if readme content is provided in site config and no other home items are found.
+  if (type === 'home' && readMeContent) {
+    config.content = readMeContent;
+    config.type = 'md';
+  }
+
+  return config;
+};
+
+/**
+* Build out the page config for the top level link.
+*/
+const getLinkPageConfig = (link, pageConfig, siteConfig) => {
+  let pages = [];
+  let type;
+
+  // If a link has more than one type, we'll have to build out at least three page configs.
+  if (link.pageTypes.length > 1) {
+    // Do not sort link types. It's up to the consumer to order that array.
+    pages = link.pageTypes.reduce((acc, pageType) => {
+      const generatedPageConfig = getPageConfig(pageType, `/${kebabCase(pageType)}`, pageConfig[pageType], pageType, siteConfig, routeImporter);
+      acc.push(generatedPageConfig);
+      return acc;
+    }, []);
+  } else {
+    // If there is only one type, just grab the first one.
+    [type] = link.pageTypes;
+    pages = pageConfig[type];
+  }
+
+  const generatedPageConfig = getPageConfig(link.text, link.path, pages, type, siteConfig);
+
+  return [generatedPageConfig];
+};
+
+/**
+ * Generates the content configuration used to populate the terra-dev-site navigation components.
+ */
+const generateContentConfig = (siteConfig, pageConfig) => {
+  if (!pageConfig) {
+    return undefined;
```

Comment:
Should we return `{ content: {}, menuItems: {} }` instead to prevent the error you seen when grabbing a key on undefined? 

I could see this possibly being seen here: https://github.com/cerner/terra-dev-site/pull/175/files#diff-f443b80960106092c69c1e822a17a01fR71

Reply:
If page config is not defined, i'm fine with errors.